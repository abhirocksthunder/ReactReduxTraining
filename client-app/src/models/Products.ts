export interface Rating {
    rate: number;
    count: number;
}

export interface Products {
    id: number;
    title: string;
    price: string;
    description: string;
    category: string;
    image: string;
    rating: Rating;
}