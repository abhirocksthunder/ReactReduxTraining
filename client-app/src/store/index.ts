import { configureStore } from "@reduxjs/toolkit";
import weatherReducer from "./weatherStore";
import movieReducer from "./moviesStore";
import productReducer from "./productsStore";

export default configureStore({
  reducer: {
    weather: weatherReducer,
    movies: movieReducer,
    products: productReducer,
  },
});
