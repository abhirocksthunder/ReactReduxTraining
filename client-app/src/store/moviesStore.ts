import { createSlice } from "@reduxjs/toolkit";
import { Movies } from "../models/Movies";

interface MoviesList{
    movies: Movies[]
}

const initialState = {
    movies : []
} as MoviesList

const moviesSlicer = createSlice({
    name:'movies',
    initialState,
    reducers:{
        getNewReleases(state,action){
            return{
                ...state,
                movies: action.payload
            }
        }
    }
})

export const moviesActions = moviesSlicer.actions;

export default moviesSlicer.reducer;