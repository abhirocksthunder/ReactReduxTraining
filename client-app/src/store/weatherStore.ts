import { createSlice } from "@reduxjs/toolkit";
import { Weather } from "../models/Weather";

interface WeatherData {
    Weather: Weather[];
    loading: boolean;
    value: string;
    results: Weather[];
    searchLabel: string;
  }
  
  const initialState = {
    Weather: [],
    loading: false,
    value: "",
    results: [],
    searchLabel: '',
  } as WeatherData;
  
  const weatherSlicer = createSlice({
    name: "weather",
    initialState: initialState,
    reducers: {
      getInitialData(state, action) {
        return {
          ...state,
          Weather: action.payload,
          results: action.payload,
          loading: false,
        };
      },
      startSearch(state, action) {
        return {
          ...state,
          value: action.payload,
          loading: true,
        };
      },
      finishSearch(state, action) {
        return {
          ...state,
          results: action.payload.results,
          loading: false,
          searchLabel: action.payload.searchLabel
        };
      },
      updateSelection(state, action) {
        return {
          ...state,
          results: action.payload,
        };
      },
      clearSearch(state) {
        return {
          ...state,
          loading: false,
          results:[],
          searchLabel:''
        };
      },
    },
  });
  
  export const WeatherActions = weatherSlicer.actions;

  export default weatherSlicer.reducer; 