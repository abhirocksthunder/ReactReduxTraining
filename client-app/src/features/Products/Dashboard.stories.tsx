import React from 'react';
import Dashboard from './Dashboard';
import { storiesOf } from '@storybook/react';
import Provider from '../provider';




storiesOf('Products', module)
    .addDecorator(story => <Provider story={story()} />)
    .add('default', () => <Dashboard />);
