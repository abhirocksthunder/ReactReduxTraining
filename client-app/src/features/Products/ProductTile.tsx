import React from "react";
import { Card } from "react-bootstrap";
import { Products } from "../../models/Products";

interface Props{
    product: Products
}


export default function ProductTile({product}: Props) {
    return (
        <Card>
            <Card.Img variant="top" src={product.image} style={{width: '100%', height:'15rem', objectFit:'cover'}}/>
            <Card.Body>
                <Card.Title style={{minHeight : '5rem', height:'5rem'}} title={product.title}>{product.title.substring(0,50)}...</Card.Title>
                <Card.Text style={{float : 'right'}}>
                    $ {product.price}
                </Card.Text>
            </Card.Body>
        </Card>
    )
}