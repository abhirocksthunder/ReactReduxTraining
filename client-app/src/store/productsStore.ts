import { createSlice } from "@reduxjs/toolkit";
import { Products } from "../models/Products";

interface ProductsList{
    products: Products[]
}

const initialState = {
    products : []
} as ProductsList

const productsSlicer = createSlice({
    name:'products',
    initialState,
    reducers:{
        getProducts(state,action){
            return{
                ...state,
                products: action.payload
            }
        }
    }
})

export const productsActions = productsSlicer.actions;

export default productsSlicer.reducer;