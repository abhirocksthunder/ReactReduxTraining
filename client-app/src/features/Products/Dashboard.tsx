import { useEffect } from 'react';
import { Col, Row, Spinner } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';
import { Container } from 'semantic-ui-react';
import { Products } from '../../models/Products';
import { productsActions } from '../../store/productsStore';
import agent from '../api/agent';
import Cards from '../Movies/Cards';

export default function Dashboard() {


    const dispatch = useDispatch();

    const fetchProducts = async () => {
        const response = await agent.products.list();
        dispatch(productsActions.getProducts(response));
    }

    useEffect(() => {

        fetchProducts();
    }, []);
    const products: Products[] = useSelector((state: any) => state.products.products);

    if (products.length === 0) return (
        <div style={{ textAlign: 'center', top: '20%' }}>
            <Spinner animation="border" size="sm" />
        </div>)

    return (
        <Container>
            <div>
                New Products
            </div>

            <Row xs={1} md={2} lg={4} className="g-4">
                {Array.from(products).sort((a, b) => parseFloat(a.price) - parseFloat(b.price)).map((product, idx) => (
                    <Col>
                        <Cards {...product} />
                    </Col>
                ))}
            </Row>

        </Container>

    )
}