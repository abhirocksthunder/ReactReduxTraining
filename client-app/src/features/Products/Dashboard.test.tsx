import Products from './Dashboard';
import { mount } from 'enzyme';
import { Provider } from "react-redux";
import store from '../../store';

describe('Product Component', ()=>{
    
    let wrapper: any;    
   
    describe('with props',()=>{
        beforeEach(()=>{
            
            wrapper = mount(<Provider store={store}><Products /></Provider>)
        });

        it('+++ render the connected(SMART) component', () => {
            expect(wrapper.find(Products).length).toMatchSnapshot();
         });
    })
});

// import Products from './Dashboard';
// import { mount } from 'enzyme';
// import configureStore from 'redux-mock-store';
// import { ProductsList } from "../../store/productsStore";
// import { Provider } from "react-redux";



// describe('Product Component', ()=>{
//     const initialState = {
//         products : []
//     } as ProductsList
//     let store, wrapper: any;
//     //let wrapper: any;
//     const mockStore = configureStore();

//     describe('with props',()=>{
//         beforeEach(()=>{
//             store = mockStore(initialState);
//             wrapper = mount(<Provider store={store}><Products /></Provider>)
//         });

//         it('+++ render the connected(SMART) component', () => {
//             expect(wrapper.find(Products).length).toMatchSnapshot();
//          });
//     })
// });