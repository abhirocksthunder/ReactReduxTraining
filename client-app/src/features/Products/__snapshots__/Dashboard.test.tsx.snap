// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Component with props +++ render the connected(SMART) component 1`] = `1`;
